ckan/logic/__init__.py

```diff
@@ -394,20 +404,32 @@ def get_action(action):
     # Then overwrite them with any specific ones in the plugins:
     resolved_action_plugins = {}
     fetched_actions = {}
+    chained_actions = defaultdict(list)
     for plugin in p.PluginImplementations(p.IActions):
         for name, auth_function in plugin.get_actions().items():
-            if name in resolved_action_plugins:
+            if _is_chained_action(auth_function):
+                chained_actions[name].append(auth_function)
+            elif name in resolved_action_plugins:
                 raise NameConflict(
                     'The action %r is already implemented in %r' % (
                         name,
                         resolved_action_plugins[name]
                     )
                 )
-            resolved_action_plugins[name] = plugin.name
-            # Extensions are exempted from the auth audit for now
-            # This needs to be resolved later
-            auth_function.auth_audit_exempt = True
-            fetched_actions[name] = auth_function
+            else:
+                resolved_action_plugins[name] = plugin.name
+                # Extensions are exempted from the auth audit for now
+                # This needs to be resolved later
+                auth_function.auth_audit_exempt = True
+                fetched_actions[name] = auth_function
+    for name, func_list in chained_actions.iteritems():
+        if name not in fetched_actions:
+            raise NotFound('The action %r is not found for chained action' % (
+                name))
+        for func in func_list:
```

Comment:
I think this should be `for func in reversed(func_list)`

For all other interfaces we have a "first plugin wins" pattern, so the first plugin declaring a chained action in the plugin list should be the first called for that action.